### PR TITLE
Test overlayfunc

### PR DESCRIPTION
**Pull request by shushengfuqin**
### Summary 
Overall, there are new tests for each function. Also, for each function, hook, and const, the types are annotated. 

### Details

- Overlay: Some functions and the hooks are moved into the util/overlayfunc/overlayfunc.tsx. The test file is overlay.test

- PreviewOverlay: removed hook functions into util/previewOverlayFun/previewOverlayfunc.tsx

- Types for ActionBox, ActionSequenceBox and ApparatusListBox aren't finalized yet. However, they do not affect how the webpage functions. 

### How to test?
- To run the test and the test coverage, please check the readme file.
- Check if all functionalities still run the way they are supposed to.

### Checks 
- [x] Tested Changes
- [x] Stakeholder Approval

### CODE REVIEW

File: pages/experience.tsx [hunk 1]
```diff
@@ -100,8 +100,8 @@ userld]); */
 }
 
 Experience.getInitialProps = ({ query }) => {
-  const dataId = query.dataId;
```
Comment (Ysabz) [inline pages/experience.tsx:103]: why this syntax is changed ?

**Review by tashfia98 — COMMENTED** [review]
Great work! The branch seems to be working fine on my end

File: src/components/Overlay/Overlay.tsx [hunk 2]
```diff
@@ -146,13 +128,13 @@ function Overlay({ userId, experienceData }): JSX.Element {
...
         <OverlayShown>
-          <PreviewOverlay actionList = {actionList} />
+          <PreviewOverlay actionList={actionList} />
         </OverlayShown>
       )}
     </OverlayRoot>
   );
 
-  function checkIfMetaExists(): any {
+  function checkIfMetaExists(): unknown {
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/components/Overlay/Overlay.tsx]: I'm not sure what is the type of it, Therefore they are either any. But if you are sure that they are apparatusMetadata type. They are good

File: src/test/loginCookies.test.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+describe("test checkinloggedin() with no login cookie", () => {
+
+  it("cookies.get", () => {
+    expect(checkIfLoggedIn()).toBe(false);
+  });
+});
+
+describe("test checkinloggedin() with cookie set", () => {
+  //   jest.mock("universal-cookie", () => ({
```
[the remark this answers is not in the document]
Comment (shushengfuqin) [reply src/test/loginCookies.test.tsx:15]: we can remove it.

File: pages/experience.tsx (hunk 1 above)
Comment (shushengfuqin) [reply pages/experience.tsx:71]: it's the same thing. It was probably auto change by es-lint

**Pull request merged**